### ISSUE TITLE
Object shadows (in ver 100)

### ISSUE THREAD

**Issue by maemaemae3** [IceCreamYou/THREE.Terrain#15]
Hi, this lib is very easy to use. Thanks a lot!
However, it's not able to recieve shadows in any materials.

Here is the screenshot.(the under layer which casting shadow correctly is MeshLambertMaterial I implemented.)
![image](https://user-images.githubusercontent.com/6958342/52035448-e7ac4300-256d-11e9-8bc3-46aef276e6b9.png)

I implemented like this:
```javascript

// Generate a terrain
const map_gras = texLoader.load('./img/snow.jpg');
const map_sand = texLoader.load('./img/sand.jpg');
const map_ston = texLoader.load('./img/ston.jpg');
const map_snow = texLoader.load('./img/snow.jpg');
tex_load_manager.onLoad = function () {
	const blend = THREE.Terrain.generateBlendedMaterial([
		{ texture: map_sand },
		{ texture: map_gras, levels: [-80, -35, 20, 50] },
		{ texture: map_ston, levels: [20, 50, 60, 85] },
		{ texture: map_snow, glsl: '1.0 - smoothstep(65.0 + smoothstep(-256.0, 256.0, vPosition.x) * 10.0, 80.0, vPosition.z)' },
		{ texture: map_ston, glsl: 'slope > 0.7853981633974483 ? 0.2 : 1.0 - smoothstep(0.47123889803846897, 0.7853981633974483, slope) + 0.2' }, // between 27 and 45 degrees
	]);

	let xS = 128, yS = 128;
	let terrainScene = THREE.Terrain({
		easing: THREE.Terrain.Linear,
		frequency: 2.5,
		heightmap: THREE.Terrain.DiamondSquare,
		material: blend,
		maxHeight: 100,
		minHeight: -100,
		steps: 1,
		useBufferGeometry: false,
		xSegments: xS,
		xSize: 1024,
		ySegments: yS,
		ySize: 1024,
	});
	terrainScene.receiveShadow = true;
	// Assuming you already have your global scene, add the terrain to it
	scene.add(terrainScene);
}.bind(this);
```

I'm using threejs version 100, but it seems not working since ver 91.
It seems related to #6 but I'm new at GLSL so it's bit confusing to investigate this.

would you henp me?

**Comment by IceCreamYou** [comment]
From looking at the [migration guide](https://github.com/mrdoob/three.js/wiki/Migration-Guide#r94--r95) it appears that the problem is likely that [this line](https://github.com/IceCreamYou/THREE.Terrain/blob/gh-pages/src/materials.js#L137) needs to be updated as `lights_pars_maps` was renamed to `envmap_physical_pars_fragment`. Maybe that will be enough. However the materials system has been restructured somewhat since the last time I looked and it's very possible that further updates would be required.

If you're willing to dive in a bit, feel free to try that change. Otherwise, I would recommend using r94 or earlier if you want this to work, for now. Unfortunately I don't expect to have the time soon to dig into it much further.

**Comment by maemaemae3** [comment]
Thank you for the response.
I changed `lights_pars_maps` to  `envmap_physical_pars_fragment`, but it has no change.
I tested it on ver r91.

Is shadowmap enabled default in blended texture shader material? Or is there any options like recieveShadow?
Anyway, I will continue investigations!

**Comment by IceCreamYou** [comment]
The `envmap_physical_pars_fragment` change would only work on r95+ and would probably throw an error in earlier versions. You'd also want to confirm that your webpage is using a version with the change in it - for example if you're using a minified build you'd either need to change the minified file or compile the library after changing a source file.

The material shouldn't be any different than other materials in terms of receiving shadows. Obviously the demo receives shadows the terrain casts onto itself, if that's a helpful reference.

**Comment by maemaemae3** [comment]
Yes, I tested with below version combination.
- r91 and `lights_pars_maps`
- r101 and `envmap_physical_pars_fragment`

I know that demo receives shadows the terrain casts onto itself, but I want to recieve shadow from object.
(For example, in the THREE.Terrain demo, I want to cast the shadow from each trees)

**Comment by IceCreamYou** [comment]
I'm sorry, I don't think I can help further in the time I have available.

**Comment by maemaemae3** [comment]
No problem. Thank you very much for giving me many advices!
I keep trial and error.

**Comment by IceCreamYou** [comment]
I haven't specifically tried it, but I think this should be fixed in v1.6.1+ thanks to the improved way the textures are generated. Please feel free to reopen if that's not the case.
